src/main/java/core/basesyntax/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,117 @@
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public Page<OrderDto> getOrders(Pageable pageable, User user) {
+        return orderRepository.findAllByUser(user, pageable)
+                .map(orderMapper::toDto);
+    }
+
+    @Override
+    @Transactional
+    public OrderDto createOrder(CreateOrderRequestDto requestDto, User user) {
+        ShoppingCart shoppingCart = shoppingCartRepository.getByUser(user);
+        if (shoppingCart.getCartItems().isEmpty()) {
+            throw new ItemsNotFoundException("No items found in shopping cart to create order");
+        }
+        Set<OrderItem> orderItems = shoppingCart.getCartItems().stream()
+                .map(orderMapper::orderItemFromCartItem)
+                .collect(Collectors.toSet());
+        cartItemRepository.removeAllByShoppingCart(shoppingCart);
+
+        BigDecimal total = orderItems.stream()
+                .map(orderItem -> orderItem.getPrice()
+                        .multiply(BigDecimal.valueOf(orderItem.getQuantity())))
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+
+        Order order = Order.builder()
+                .shippingAddress(requestDto.shippingAddress())
+                .orderDate(LocalDateTime.now())
+                .status(Order.Status.CREATED)
+                .orderItems(orderItems)
+                .total(total)
+                .user(user)
+                .build();
+        orderRepository.save(order);
+
+        orderItems.forEach(orderItem -> orderItem.setOrder(order));
+        orderItemRepository.saveAll(orderItems);
+
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public OrderDto getOrderById(Long orderId, User user) {
+        return orderMapper.toDto(orderRepository.findByIdAndUser(orderId, user).orElseThrow(() ->
+                new EntityNotFoundException("Can't find order by id. Id: " + orderId)));
+    }
+
+    @Override
+    public Page<OrderItemDto> getOrderItems(Long orderId, Pageable pageable, User user) {
+        Order order = orderRepository.findByIdAndUser(orderId, user).orElseThrow(() ->
+                new EntityNotFoundException("Can't find order by id. Id: " + orderId));
+        Page<OrderItem> orderItems = orderItemRepository
+                .findAllByOrder(order, pageable);
+        List<OrderItemDto> orderItemDtos = orderItems
+                .stream()
+                .map(orderMapper::toOrderItemDto)
+                .toList();
+        return new PageImpl<>(orderItemDtos, pageable, orderItems.getTotalElements());
+    }
+
+    @Override
+    public OrderItemDto getOrderItemById(Long orderId, Long orderItemId, User user) {
+        if (orderRepository.existsByIdAndUser(orderId, user)) {
```

Comment:
you can use `OrderItem item = itemRepository.findByIdAndOrderIdAndUserId(orderItemId, orderId, userId)`

Reply:
order item has no info about user